ml-prims/src/common/buffer_base.hpp

```diff
@@ -88,6 +88,20 @@ class buffer_base
     size_type                           _size;
     size_type                           _capacity;
     value_type*                         _data;
+    void set_stream( cudaStream_t stream )
+    {
+        cudaEvent_t event;
```

Comment:
@jirikraus in case the user passes the same stream as the current `_stream`, then there'd still be a `cudaStreamWaitEvent` called, right?

Reply:
Good point. I added a check that the passed in stream is actually different.